### PR TITLE
backend/internetarchive: add support for Metadata

### PR DESCRIPTION
**Pull request by Lesmiscore**
#### What is the purpose of this change?

This PR adds support for viewing metadata in IA from rclone.
~~I can't figure out how to implement writer part, so I've only implemented metadata read part (fs.Metadataer).~~

#### Was the change discussed in an issue or in the forum before?

None.

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/internetarchive/internetarchive.go [hunk 1]
```diff
@@ -226,7 +308,10 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
 	}
 	f.setRoot(root)
 	f.features = (&fs.Features{
-		BucketBased: true,
+		BucketBased:   true,
+		ReadMetadata:  true,
+		WriteMetadata: true,
```
Comment (ncw) [inline backend/internetarchive/internetarchive.go:313]: This should be false if we can't write metadata

File: backend/internetarchive/internetarchive.go [hunk 2]
```diff
@@ -226,7 +308,10 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
...
+		UserMetadata:  true,
```
Comment (ncw) [inline backend/internetarchive/internetarchive.go:314]: Also this if we can't write user metadata

**Review by ncw — COMMENTED** [review]
This looks very useful :-)

There is also a table in `overview.md` about what support backends have for metadata - you can add internetarchive there as `R` I think.

A full `RWU` support would be nice, but `R` support is very useful.

Do the metadata integration tests pass in the backend tests?

File: backend/internetarchive/internetarchive.go (hunk 1 above)
Comment (Lesmiscore) [reply backend/internetarchive/internetarchive.go:313]: We can write/update it (like we do for setting mtime) but just not implemented

Comment (Lesmiscore) [reply backend/internetarchive/internetarchive.go:313]: So I was expecting some interface to add metadata, but fs.ObjectInfo in Update() has fs.Metadata implemented?

**Comment by Lesmiscore** [comment]
```
$ go test -v -timeout 114514h -run TestIntegration/FsMkdir/FsPutFiles/ObjectMetadata backend/internetarchive/internetarchive_test.go -args --remote TestIA:lesmi-rclone-test/
=== RUN   TestIntegration
    fstests.go:451: Using remote "TestIA:lesmi-rclone-test/"
=== RUN   TestIntegration/FsMkdir
=== RUN   TestIntegration/FsMkdir/FsPutFiles
2022/07/07 00:26:40 WARNING: file name.txt: the reserved metadata key rclone-test is about to set
=== RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectMetadata
=== RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectMetadata/mtime
--- PASS: TestIntegration (481.70s)
    --- PASS: TestIntegration/FsMkdir (479.92s)
        --- PASS: TestIntegration/FsMkdir/FsPutFiles (463.28s)
            --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectMetadata (172.37s)
                --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectMetadata/mtime (172.15s)
PASS
ok      command-line-arguments  481.705s
```
Now it passes metadata test

File: backend/internetarchive/internetarchive.go (hunk 1 above)
Comment (Lesmiscore) [reply backend/internetarchive/internetarchive.go:326]: Thanks, it's reflected

**Review by ncw — APPROVED** [review]
This looks good now - thank you :-)

I ran the integration tests and they passed fine.

I'll merge this for v1.59.

**Pull request merged**